tests/spec_tests/type_constraint_tests/group_by.yml

```diff
@@ -2,35 +2,112 @@ variables:
   bool: &bool ["BOOL", "NULL", "MISSING"]
   document: &document ["DOCUMENT", "NULL", "MISSING"]
   numerics: &numerics ["INT", "LONG", "DOUBLE", "DECIMAL", "NULL", "MISSING"]
-
+  selfComparisonValidTypes: &selfComparisonValidTypes
+    - { "arg1": *numerics }
+    - { "arg1": [ "STRING", "NULL", "MISSING" ] }
+    - { "arg1": *document }
+    - { "arg1": [ "ARRAY", "NULL", "MISSING" ] }
+    - { "arg1": [ "BINDATA", "NULL", "MISSING" ] }
+    - { "arg1": [ "UNDEFINED", "NULL", "MISSING" ] }
+    - { "arg1": [ "OBJECTID", "NULL", "MISSING" ] }
+    - { "arg1": *bool }
+    - { "arg1": [ "DATE", "NULL", "MISSING" ] }
+    - { "arg1": [ "NULL", "MISSING" ] }
+    - { "arg1": [ "REGEX", "NULL", "MISSING" ] }
+    - { "arg1": [ "SYMBOL", "NULL", "MISSING" ] }
+    - { "arg1": [ "TIMESTAMP", "NULL", "MISSING" ] }
+    - { "arg1": [ "MINKEY", "NULL", "MISSING" ] }
+    - { "arg1": [ "MAXKEY", "NULL", "MISSING" ] }
 
 tests:
+  # Note: Non-distinct ADD_TO_ARRAY has no type constraints
+  - description: Distinct ADD_TO_ARRAY argument must be self-comparable
+    skip_reason: "SQL-2618: enable document and array self-comparison"
```

Comment:
I'll unskip all of these as part of this PR as soon as #13 is merged, which should be soon!